dashboard/src/app/(authenticated)/(diagram)/components/Modals/NodeRun/NodeRun.tsx

```diff
@@ -0,0 +1,80 @@
+import { FC, ReactNode } from 'react'
+import { Dialog, DialogContent, DialogHeader, DialogTitle } from '@/components/ui/dialog'
+import { Accordion } from '@/components/ui/accordion'
+
+import { Modal, NodeRuns } from '../../../context'
+import { useModal } from '../../../hooks/useModal'
+import { statusColors } from '../../../wfRun/[...ids]/components/Details'
+import { TaskLink } from '../../NodeTypes/Task/TaskDetails'
+import { AccordionComponents, AccordionConentType } from './AccordionContent'
+import { NodeRun as Node } from 'littlehorse-client/proto'
+
+export const getNodeType = (node: Node) => {
+  if (node['task'] !== undefined) return 'TASK'
+  if (node['externalEvent'] !== undefined) return 'EXTERNAL_EVENT'
+  if (node['waitThreads'] !== undefined) return 'WAIT_FOR_THREADS'
+  if (node['sleep'] !== undefined) return 'SLEEP'
+  if (node['userTask'] !== undefined) return 'USER_TASK'
+  if (node['startThread'] !== undefined) return 'START_THREAD'
+  if (node['throwEvent'] !== undefined) return 'THROW_EVENT'
+}
```

Comment:
This function already exists on `extractNodes.ts` please reuse.

Reply:
fixed